packages/java/endpoint/src/test/java/dev/hilla/HillaStatsTest.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2000-2023 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package dev.hilla;
+
+import com.vaadin.flow.internal.UsageStatistics;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertNull;
+
+public class HillaStatsTest {
+    private ClassLoader oldContextClassLoader;
+
+    @Rule
+    public TemporaryFolder temporary = new TemporaryFolder();
+
+    @Before
+    public void rememberContextClassLoader() throws Exception {
+        oldContextClassLoader = Thread.currentThread().getContextClassLoader();
+        UsageStatistics.resetEntries();
+        fakeHilla(false);
```

Comment:
It seems this makes the last test fail, so I suggest to manually call it in each the test:
```suggestion
```

Reply:
Actually I think the problematic part was the memoization of `Platform.hillaVersion`, since that value depends on the context classloader. I've added a code which erases this memoization, it fixed the test.